### PR TITLE
Evaporation model refactor

### PR DESCRIPTION
**Pull request by mschreter**
This PR is part of a comprehensive refactoring (referencing #213) of the evaporation operations. It introduces an `EvaporationModelBase` class and derived classes. The interface to the actual `EvaporationOperation` is still work in progress.

### CODE REVIEW

File: include/meltpooldg/evaporation/evaporation_model_base.hpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * ---------------------------------------------------------------------*/
+#pragma once
+
+namespace MeltPoolDG::Evaporation
+{
+  class EvaporationModelBase
+  {
+  public:
+    EvaporationModelBase() = default;
```
Comment (peterrum) [inline include/meltpooldg/evaporation/evaporation_model_base.hpp:13]: Needed?

File: source/evaporation/evaporation_operation.cpp [hunk 2]
```diff
@@ -525,15 +527,13 @@ namespace MeltPoolDG::Evaporation
...
-                 T, pressure_constant, temperature_constant, boiling_temperature) *
-               std::sqrt(Cm / T) :
-             0.0;
+    EvaporationModelRecoilPressure<dim> evapor_model(
+      boiling_temperature,
+      pressure_constant,
+      temperature_constant,
+      evaporation_data.evaporative_mass_flux_scale_factor);
+
+    return evapor_model.local_compute_evaporative_mass_flux(T);
```
Comment (peterrum) [inline source/evaporation/evaporation_operation.cpp:536]: I guess later on you plan to make the evaporation model to class member (which is the reason for the base class).

**Review by peterrum — APPROVED** [review]
Looks good to me. Maybe I could ask you for some documentation, e.g., what does `local_compute_evaporative_mass_flux()` do (input, output) and how do the actual implementations differ from each other + reference to literature. But fee free to do these in a follow-up PR(s)!

File: source/evaporation/evaporation_operation.cpp (hunk 2 above)
Comment (mschreter) [reply source/evaporation/evaporation_operation.cpp:536]: Exactly.

File: include/meltpooldg/evaporation/evaporation_model_base.hpp (hunk 1 above)
Comment (mschreter) [reply include/meltpooldg/evaporation/evaporation_model_base.hpp]: nope.

**Comment by mschreter** [comment]
Thanks for reviewing this PR. I've added the documentation. From my side this PR is ready :-), let's wait what CI has to say ;-).

File: include/meltpooldg/evaporation/evaporation_model_base.hpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * ---------------------------------------------------------------------*/
+#pragma once
+
+namespace MeltPoolDG::Evaporation
+{
+  class EvaporationModelBase
+  {
+  public:
+    virtual double
+    local_compute_evaporative_mass_flux(const double T) = 0;
```
Comment (peterrum) [inline include/meltpooldg/evaporation/evaporation_model_base.hpp:14]: Maybe also a comment here, please. Else the docu is :1st_place_medal:

**Review by peterrum — APPROVED** [review]

**Pull request merged**